### PR TITLE
Dependabot roundup

### PR DESCRIPTION
**Pull request by dirkgr**
This combines a bunch of recent dependabot changes. There are some more, but they did not merge cleanly, so I didn't bother. Dependabot will reproduce them if necessary.

### CODE REVIEW

**Review by codeviking — APPROVED** [review]

File: api/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 git+git://github.com/allenai/allennlp.git@v2.1.0
 git+git://github.com/allenai/allennlp-models.git
```
Comment (epwalsh) [inline api/requirements.txt:2]: Why do we have allennp pinned to a release and allennlp-models pinned to nothing?

Comment (dirkgr) [reply api/requirements.txt:2]: I don't remember. But we could pin both to the latest to get around this.

File: api/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-git+git://github.com/allenai/allennlp.git@v2.1.0
-git+git://github.com/allenai/allennlp-models.git
+git+git://github.com/allenai/allennlp.git@v2.3.1
+git+git://github.com/allenai/allennlp-models.git@v2.3.1
```
Comment (epwalsh) [inline api/requirements.txt:2]: ?

```suggestion
git+git://github.com/allenai/allennlp-models.git@v2.3.0
```

Comment (dirkgr) [reply api/requirements.txt:2]: Oh, I didn't even check. I thought you always ship those in tandem.

Comment (epwalsh) [reply api/requirements.txt]: We decided a while ago that we could do patch releases in the core repo when needed without a corresponding models release

**Pull request merged**